java/vector/src/main/java/org/apache/arrow/vector/BaseValueVector.java

```diff
@@ -122,12 +122,17 @@ private static long roundUp8(long size) {
     return ((size + 7) / 8) * 8;
   }
 
+  /* round up bytes for the validity buffer for the given valueCount */
+  private long roundUp8ForValidityBuffer(int valueCount) {
+    return (valueCount + 63) / 64 * 8;
```

Comment:
can this be replaced with:
((valueCount + 63) >> 6) << 3

Reply:
Sure, good point.

Reply:
Actually would just shifting left by 3?

Reply:
Nope, in that way there is no guarantee that the result will be multiple of 8.
For example:
(15 + 63) >> 3 = 9 which is not correct.

Reply:
What and mask out the least significant bits?  Does that provide better performance or is it the same?

Reply:
What do you mean by 'mask out least significant bits'?

Reply:
Something like: ((15+63) >> 3) & 0xfffffff8

Reply:
The two have almost the same performance(3804 vs 3797)

Reply:
no need to this, I think they should be the same.